openpathsampling/volume.py

```diff
@@ -10,12 +10,24 @@
 # TODO: Make Full and Empty be Singletons to avoid storing them several times!
 
 def join_volumes(volume_list):
-    volume = None
+    """
+    Make the union of a list of volumes. (Useful shortcut.)
+
+    Parameters
+    ----------
+    volume_list : list of Volume
+        the list to be joined together
+
+    Returns
+    -------
+    UnionVolume 
+        the union of the elements of the list, or EmptyVolume if list is
+        empty
+    """
+    volume = EmptyVolume()
+    # EmptyVolume is smart and knows its OR just takes the other
     for vol in volume_list:
-        if volume is None:
-            volume = vol
-        else:
-            volume = volume | vol
+        volume = volume | vol
```

Comment:
I just discovered a great website about python that covers lots of features and great packages.

http://people.duke.edu/~ccc14/sta-663/

There the function 'reduce' was mentioned what interatively applies a function reusing the result. While `map` applies each element to the function, reduce takes two elements and uses the result and the following element again. 
So you could write your function as
```
return reduce(lambda x,y : x | y, volume_list)
```
or 
```
return reduce(lambda x,y : x | y, [EmptyVolume()] + volume_list)
```
if you want to allow for empty lists. The lambda could be replaced using the `operator` module
```
import operator as op

return reduce(op.or_, [EmptyVolume()] + volume_list)
```